### PR TITLE
Move dht tests from js-ipfs-api to interface-ipfs-core

### PR DESCRIPTION
**Pull request by VictorBjelkholm**
Needs https://github.com/ipfs/interface-ipfs-core/pull/36

### CODE REVIEW

**Comment by diasdavid** [comment]
Needs:
 - [ ] https://github.com/ipfs/interface-ipfs-core/pull/36#issuecomment-247822522

**Comment by diasdavid** [comment]
interface-ipfs-core 0.18.3 released

**Comment by VictorBjelkholm** [comment]
current status in CI:
```
Message:
    Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
Stack:
SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
```

Seems linting in interface-ipfs-core runs in different env than the tests for js-ipfs-api.

File: test/interface-ipfs-core/dht.spec.js [hunk 1]
```diff
@@ -1,65 +1,18 @@
 /* eslint-env mocha */
-/* globals apiClients */
-'use strict'
```
Comment (dignifiedquire) [inline test/interface-ipfs-core/dht.spec.js]: you shouldn't delete use strict

**Comment by diasdavid** [comment]
all right :) thank you @VictorBjelkholm

**Pull request merged**